### PR TITLE
Add com.gamestarmechanic.app

### PR DESCRIPTION
**Pull request by dylanmccall**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions.
  - This application uses an old version of Electron which requires X11. The required permissions are otherwise quite trivial.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
  - We are publishing on Flathub with permission of the upstream developers.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
  - The application ID was originally `com.gamestarmechanic`. It was changed to `com.gamestarmechanic.app` for this work.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flatpak.org/en/latest/first-build.html

### CODE REVIEW

File: com.gamestarmechanic.app.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+app-id: com.gamestarmechanic.app
```
Comment (hfiguiere) [inline com.gamestarmechanic.app.yaml:2]: it should be `com.gamestarmechanic.gamestarmechanic`. `app` isn't an application name.

File: com.gamestarmechanic.app.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+modules:
+  - name: gamestar-mechanic
+    buildsystem: simple
+    build-options:
+      env:
+        XDG_CACHE_HOME: /run/build/gamestar-mechanic/flatpak-node/cache
+    sources:
+      - type: git
+        url: https://github.com/dylanmccall/gamestar-mechanic-linux.git
+        branch: main
```
Comment (hfiguiere) [inline com.gamestarmechanic.app.yaml:36]: NEVER a git branch.

File: com.gamestarmechanic.app.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      # Tweak desktop file values
+      - desktop-file-edit --set-key="Exec" --set-value="gamestarmechanic-wrapper %u" ${FLATPAK_DEST}/share/applications/com.gamestarmechanic.app.desktop
+  - name: libpepflashplayer
+    buildsystem: simple
+    sources:
+      - type: script
+        dest-filename: apply_extra
+        commands:
+          - mkdir plugins
+          - tar -zx -f flash_player_ppapi_linux.x86_64.tar.gz -C plugins libpepflashplayer.so
```
Comment (hfiguiere) [inline com.gamestarmechanic.app.yaml:58]: so how is this gonna work on `aarch64`.

File: com.gamestarmechanic.app.yaml (hunk 1 above)
Comment (dylanmccall) [reply com.gamestarmechanic.app.yaml:2]: Oh, sure, that makes sense to me! Should I make a new PR with a different source branch (and these other changes)?

File: com.gamestarmechanic.app.yaml (hunk 2 above)
Comment (dylanmccall) [reply com.gamestarmechanic.app.yaml]: Oh, embarrassing! I forgot to push all of my recent changes there, but indeed I did forget to remove the `branch` part :b Importantly, the url also changes to https://github.com/endlessm/gamestar-mechanic-linux.git in the newest commit.

File: com.gamestarmechanic.app.yaml (hunk 3 above)
Comment (dylanmccall) [reply com.gamestarmechanic.app.yaml:62]: Thank you for spotting that! The newest commit, alas, limits builds to x86_64.

File: com.gamestarmechanic.app.yaml (hunk 1 above)
Comment (hfiguiere) [reply com.gamestarmechanic.app.yaml]: branch names are irrelevant

**Comment by hfiguiere** [comment]
bot, build com.gamestarmechanic.gamestarmechanic

**Pull request closed without merging**